File: spaceship_crm/java8/spaceship_crm_java8_rest_jooq/src/main/java/com/spaceship/crm/service/SecurityService.java
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.service;

import com.spaceship.crm.bean.CustomUserDetails;
import com.spaceship.crm.bean.LoginResponseBean;
import com.spaceship.crm.entity.User;
import com.spaceship.crm.security.JwtService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class SecurityService {

    @Autowired
    private JwtService jwtService;

    @Autowired
    private AuthenticationManager authenticationManager;

    public Optional<LoginResponseBean> doLogin(User user) {
        UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(
                user.getUsername(),
                user.getPassword()
        );
        Authentication auth = authenticationManager.authenticate(token);
        if (auth.isAuthenticated()) {
            return Optional.of(new LoginResponseBean(
                    ((CustomUserDetails)auth.getPrincipal()).getUser(),
                    jwtService.generateToken(auth)
            ));
        }
        return Optional.empty();
    }

}

File: spaceship_crm/scala/spaceship_crm_scala_spring_rest_jpa/src/main/scala/com/spaceship/crm/security/SecurityService.scala
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.security

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.{AuthenticationManager, AuthenticationServiceException, UsernamePasswordAuthenticationToken}
import org.springframework.stereotype.Service
import com.spaceship.crm.bean.LoginResponseBean
import com.spaceship.crm.entity.User

@Service
class SecurityService(
                     @Autowired private val jwtService: JwtService,
                     @Autowired private val authenticationManager: AuthenticationManager
                     ) {
  def doLogin(user: User): LoginResponseBean = {
    val token = new UsernamePasswordAuthenticationToken(
      user.username,
      user.password
    )

    val auth = authenticationManager.authenticate(token)

    if (auth == null || !auth.isAuthenticated)
      throw new AuthenticationServiceException("Authentication exception")

    new LoginResponseBean(
      auth.getPrincipal.asInstanceOf[CustomUserDetails].user,
      jwtService.generateToken(auth)
    )
  }
}
